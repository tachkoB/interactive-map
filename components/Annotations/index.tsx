import { Annotation } from 'react-simple-maps';

interface City {
  name: string;
  coordinates: [number, number];
  dx: number;
  dy: number;
}

const cities: Array<City> = [
  {
    name: 'Zurich',
    coordinates: [8.5417, 47.3769],
    dx: 10,
    dy: 20,
  },
  {
    name: 'Berlin',
    coordinates: [13.405, 52.52],
    dx: 30,
    dy: -30,
  },
  {
    name: 'Frankfurt',
    coordinates: [8.6821, 50.1109],
    dx: -20,
    dy: -20,
  },
  {
    name: 'London',
    coordinates: [-0.1257, 51.5085],
    dx: -30,
    dy: 10,
  },
  {
    name: 'Manilla',
    coordinates: [120.9842, 14.5995],
    dx: -20,
    dy: 30,
  },
  {
    name: 'New York',
    coordinates: [-74.006, 40.7128],
    dx: 30,
    dy: 40,
  },
  {
    name: 'Taguig',
    coordinates: [121.0509, 14.5176],
    dx: 60,
    dy: -15,
  },
  {
    name: 'Tokyo',
    coordinates: [139.6503, 35.6762],
    dx: 60,
    dy: -40,
  },
  {
    name: 'Toronto',
    coordinates: [-79.3832, 43.6532],
    dx: -45,
    dy: -30,
  },
];

const Annotations = () => {
  return (
    <>
      {cities.map(({ name, coordinates, dx, dy }) => (
        <Annotation
          subject={coordinates}
          dx={dx}
          dy={dy}
          connectorProps={{
            stroke: 'orange',
            strokeWidth: 1,
            strokeLinecap: 'round',
          }}
        >
          <text
            x="-8"
            textAnchor="end"
            alignmentBaseline="middle"
            fill="orange"
          >
            {name}
          </text>
        </Annotation>
      ))}
    </>
  );
};

export default Annotations;
