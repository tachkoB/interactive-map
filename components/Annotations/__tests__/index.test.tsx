import Annotations from '../index';
import { render } from '@testing-library/react';
import { ComposableMap, ZoomableGroup } from 'react-simple-maps';

const Component = (
  <ComposableMap>
    <ZoomableGroup width={200} height={200}>
      <Annotations />
    </ZoomableGroup>
  </ComposableMap>
);

describe('Annotations component', () => {
  it('should render the component', () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it('should render the names of all cities', () => {
    const { container, getByText } = render(Component);

    expect(container).toMatchSnapshot();

    const cities = [
      'Zurich',
      'Manilla',
      'New York',
      'London',
      'Toronto',
      'Berlin',
      'Frankfurt',
      'Taguig',
      'Tokyo',
    ];

    cities.forEach((city) => {
      expect(getByText(city)).toBeInTheDocument();
    });
  });
});
