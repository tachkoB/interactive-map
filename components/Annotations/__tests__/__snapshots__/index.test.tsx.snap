// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Annotations component should render the component 1`] = `
<div>
  <svg
    class="rsm-svg "
    viewBox="0 0 800 600"
  >
    <g>
      <rect
        fill="transparent"
        height="600"
        width="800"
      />
      <g
        class="rsm-zoomable-group undefined"
        height="200"
        transform="translate(0 0) scale(1)"
        width="200"
      >
        <g
          class="rsm-annotation "
          transform="translate(379.1613148586466, 180.65323933284435)"
        >
          <path
            d="M0,0 Q20,-20 40,-40"
            fill="transparent"
            stroke="orange"
            stroke-linecap="round"
            stroke-width="1"
          />
          <text
            alignment-baseline="middle"
            fill="orange"
            text-anchor="end"
            x="-8"
          >
            Zurich
          </text>
        </g>
        <g
          class="rsm-annotation "
          transform="translate(458.8408236781645, 96.26647051721704)"
        >
          <path
            d="M0,0 Q-15,15 -30,30"
            fill="transparent"
            stroke="orange"
            stroke-linecap="round"
            stroke-width="1"
          />
          <text
            alignment-baseline="middle"
            fill="orange"
            text-anchor="end"
            x="-8"
          >
            Berlin
          </text>
        </g>
        <g
          class="rsm-annotation "
          transform="translate(394.0606523810681, 112.88868436104914)"
        >
          <path
            d="M0,0 Q12.5,10 25,20"
            fill="transparent"
            stroke="orange"
            stroke-linecap="round"
            stroke-width="1"
          />
          <text
            alignment-baseline="middle"
            fill="orange"
            text-anchor="end"
            x="-8"
          >
            Frankfurt
          </text>
        </g>
        <g
          class="rsm-annotation "
          transform="translate(369.7272164110342, 139.02079754879088)"
        >
          <path
            d="M0,0 Q15,-5 30,-10"
            fill="transparent"
            stroke="orange"
            stroke-linecap="round"
            stroke-width="1"
          />
          <text
            alignment-baseline="middle"
            fill="orange"
            text-anchor="end"
            x="-8"
          >
            London
          </text>
        </g>
        <g
          class="rsm-annotation "
          transform="translate(817.3062973457756, 287.9016848690285)"
        >
          <path
            d="M0,0 Q-50,-20 -100,-40"
            fill="transparent"
            stroke="orange"
            stroke-linecap="round"
            stroke-width="1"
          />
          <text
            alignment-baseline="middle"
            fill="orange"
            text-anchor="end"
            x="-8"
          >
            Manilla
          </text>
        </g>
        <g
          class="rsm-annotation "
          transform="translate(256.0553892434068, 200.56527690454024)"
        >
          <path
            d="M0,0 Q-15,-20 -30,-40"
            fill="transparent"
            stroke="orange"
            stroke-linecap="round"
            stroke-width="1"
          />
          <text
            alignment-baseline="middle"
            fill="orange"
            text-anchor="end"
            x="-8"
          >
            New York
          </text>
        </g>
        <g
          class="rsm-annotation "
          transform="translate(777.5368447688975, 223.19061883945625)"
        >
          <path
            d="M0,0 Q-30,12.5 -60,25"
            fill="transparent"
            stroke="orange"
            stroke-linecap="round"
            stroke-width="1"
          />
          <text
            alignment-baseline="middle"
            fill="orange"
            text-anchor="end"
            x="-8"
          >
            Taguig
          </text>
        </g>
        <g
          class="rsm-annotation "
          transform="translate(798.2011139683544, 136.41387443563735)"
        >
          <path
            d="M0,0 Q-30,20 -60,40"
            fill="transparent"
            stroke="orange"
            stroke-linecap="round"
            stroke-width="1"
          />
          <text
            alignment-baseline="middle"
            fill="orange"
            text-anchor="end"
            x="-8"
          >
            Tokyo
          </text>
        </g>
        <g
          class="rsm-annotation "
          transform="translate(172.0322126958872, 121.61828452063205)"
        >
          <path
            d="M0,0 Q22.5,15 45,30"
            fill="transparent"
            stroke="orange"
            stroke-linecap="round"
            stroke-width="1"
          />
          <text
            alignment-baseline="middle"
            fill="orange"
            text-anchor="end"
            x="-8"
          >
            Toronto
          </text>
        </g>
      </g>
    </g>
  </svg>
</div>
`;

exports[`Annotations component should render the names of all cities 1`] = `
<div>
  <svg
    class="rsm-svg "
    viewBox="0 0 800 600"
  >
    <g>
      <rect
        fill="transparent"
        height="600"
        width="800"
      />
      <g
        class="rsm-zoomable-group undefined"
        height="200"
        transform="translate(0 0) scale(1)"
        width="200"
      >
        <g
          class="rsm-annotation "
          transform="translate(379.1613148586466, 180.65323933284435)"
        >
          <path
            d="M0,0 Q20,-20 40,-40"
            fill="transparent"
            stroke="orange"
            stroke-linecap="round"
            stroke-width="1"
          />
          <text
            alignment-baseline="middle"
            fill="orange"
            text-anchor="end"
            x="-8"
          >
            Zurich
          </text>
        </g>
        <g
          class="rsm-annotation "
          transform="translate(458.8408236781645, 96.26647051721704)"
        >
          <path
            d="M0,0 Q-15,15 -30,30"
            fill="transparent"
            stroke="orange"
            stroke-linecap="round"
            stroke-width="1"
          />
          <text
            alignment-baseline="middle"
            fill="orange"
            text-anchor="end"
            x="-8"
          >
            Berlin
          </text>
        </g>
        <g
          class="rsm-annotation "
          transform="translate(394.0606523810681, 112.88868436104914)"
        >
          <path
            d="M0,0 Q12.5,10 25,20"
            fill="transparent"
            stroke="orange"
            stroke-linecap="round"
            stroke-width="1"
          />
          <text
            alignment-baseline="middle"
            fill="orange"
            text-anchor="end"
            x="-8"
          >
            Frankfurt
          </text>
        </g>
        <g
          class="rsm-annotation "
          transform="translate(369.7272164110342, 139.02079754879088)"
        >
          <path
            d="M0,0 Q15,-5 30,-10"
            fill="transparent"
            stroke="orange"
            stroke-linecap="round"
            stroke-width="1"
          />
          <text
            alignment-baseline="middle"
            fill="orange"
            text-anchor="end"
            x="-8"
          >
            London
          </text>
        </g>
        <g
          class="rsm-annotation "
          transform="translate(817.3062973457756, 287.9016848690285)"
        >
          <path
            d="M0,0 Q-50,-20 -100,-40"
            fill="transparent"
            stroke="orange"
            stroke-linecap="round"
            stroke-width="1"
          />
          <text
            alignment-baseline="middle"
            fill="orange"
            text-anchor="end"
            x="-8"
          >
            Manilla
          </text>
        </g>
        <g
          class="rsm-annotation "
          transform="translate(256.0553892434068, 200.56527690454024)"
        >
          <path
            d="M0,0 Q-15,-20 -30,-40"
            fill="transparent"
            stroke="orange"
            stroke-linecap="round"
            stroke-width="1"
          />
          <text
            alignment-baseline="middle"
            fill="orange"
            text-anchor="end"
            x="-8"
          >
            New York
          </text>
        </g>
        <g
          class="rsm-annotation "
          transform="translate(777.5368447688975, 223.19061883945625)"
        >
          <path
            d="M0,0 Q-30,12.5 -60,25"
            fill="transparent"
            stroke="orange"
            stroke-linecap="round"
            stroke-width="1"
          />
          <text
            alignment-baseline="middle"
            fill="orange"
            text-anchor="end"
            x="-8"
          >
            Taguig
          </text>
        </g>
        <g
          class="rsm-annotation "
          transform="translate(798.2011139683544, 136.41387443563735)"
        >
          <path
            d="M0,0 Q-30,20 -60,40"
            fill="transparent"
            stroke="orange"
            stroke-linecap="round"
            stroke-width="1"
          />
          <text
            alignment-baseline="middle"
            fill="orange"
            text-anchor="end"
            x="-8"
          >
            Tokyo
          </text>
        </g>
        <g
          class="rsm-annotation "
          transform="translate(172.0322126958872, 121.61828452063205)"
        >
          <path
            d="M0,0 Q22.5,15 45,30"
            fill="transparent"
            stroke="orange"
            stroke-linecap="round"
            stroke-width="1"
          />
          <text
            alignment-baseline="middle"
            fill="orange"
            text-anchor="end"
            x="-8"
          >
            Toronto
          </text>
        </g>
      </g>
    </g>
  </svg>
</div>
`;
